.ci/go-test.sh

```diff
@@ -64,16 +70,22 @@ test_coverage()
 {
 	echo "mode: atomic" > "$test_coverage_file"
 
+	users="current"
+
 	if [ $(id -u) -eq 0 ]; then
-		echo >&2 "WARNING: Already running as root so will not re-run tests as non-root user."
-		echo >&2 "WARNING: As a result, only a subset of tests will be run"
-		echo >&2 "WARNING: (run this script as a non-privileged to ensure all tests are run)."
-		users="current"
+		warn "Already running as root so will not re-run tests as non-root user."
+		warn "As a result, only a subset of tests will be run"
+		warn "(run this script as a non-privileged to ensure all tests are run)."
 	else
-		# Run the unit-tests *twice* (since some must run as root and
-		# others must run as non-root), combining the resulting test
-		# coverage files.
-		users="current root"
+		if [ -n "$KATA_DEV_MODE" ]; then
+			warn "Dangerous to set CI with KATA_DEV_MODE set"
+			warn "Not also running tests as root"
```

Comment:
can we rephrase this line - it doesn't quite make sense in my head....

Reply:
Agreed - reworded for improved clarity.